src/stateMachines/wifiNetworkConfiguration.ts

```diff
@@ -305,22 +311,22 @@ export class WiFiConfiguration {
             src: this.addWifiConfigs.bind(this),
             id: 'add-wifi-settings',
             onDone: {
-              actions: assign({ message: (context, event) => event.data }),
+              actions: assign({
```

Comment:
missing a check of the response code from the invokeWsmanCall. (see isNotWifiProfileAdded condition. the onError handler only gets invoked if http return status is not 200, but AMT also reports ReturnValue)

Reply:
Updated code as per the comment